src/main/java/com/puppycrawl/tools/checkstyle/checks/design/FinalClassCheck.java

```diff
@@ -253,14 +253,39 @@ public void leaveToken(DetailAST ast) {
      * @return true if given class should be declared as final otherwise false
      */
     private static boolean shouldBeDeclaredAsFinal(ClassDesc desc) {
-        return desc.isWithPrivateCtor()
+        return (desc.isWithPrivateCtor() || isDefaultConstructor(desc.getTypeDeclarationAst()))
                 && !(desc.isDeclaredAsAbstract()
                     || desc.isSuperClassOfAnonymousInnerClass())
                 && !desc.isDeclaredAsFinal()
                 && !desc.isWithNonPrivateCtor()
                 && !desc.isWithNestedSubclass();
     }
 
+    /**
+     * Check if class is private.
+     *
+     * @param desc description of class
+     * @return true if class is private
+     */
+    private static boolean isClassPrivate(DetailAST desc) {
+        return desc.getFirstChild().findFirstToken(TokenTypes.LITERAL_PRIVATE) != null;
+    }
+
+    /**
+     * Check it the constructor is default. default means there is no constructor
+     * defined.
+     *
+     * @param desc class to check
+     * @return boolean value for default constructor
+     */
+    private static boolean isDefaultConstructor(DetailAST desc) {
+        boolean ans = false;
```

Comment:
Please use better name for asn, I don't understand it.

Reply:
Name of method should be hasExplicitContructor, (isClassPrivate(desc)) should not be in method.

Reply:
I have write it as hasImplicitConstructor.